src/minuteman_lashup_vip_listener.erl

```diff
@@ -192,6 +198,28 @@ code_change(_OldVsn, State, _Extra) ->
 %%% Internal functions
 %%%===================================================================
 
+handle_lookup_vips(State, Names) ->
+  lists:map(fun(Name) -> handle_lookup_vip(State, Name) end, Names).
+
+handle_lookup_vip(State, {ip, IP}) ->
```

Comment:
So, this should probably short-circuit the gen_Server, and use the ets table directly from the public function. Otherwise, your performance will be bound by the gen_server synchronous answer.